README.adoc

```diff
@@ -253,23 +211,95 @@ public class Album {
+This allows you to reduce the number of round-trips between your application and Cloud Spanner.
+
+By default, Hibernate does not batch the statements that it sends to the Cloud Spanner JDBC driver.
+
+Batching can be enabled by configuring `hibernate.jdbc.batch_size` in your Hibernate configuration file:
+
+[source, xml]
+----
+<property name="hibernate.jdbc.batch_size">100</property>
+----
+
+The property is set to `100` as an example; you may experiment with the batch size to see what works best for your application.
+
+==== Query Optimization
+
+The https://cloud.google.com/spanner/docs/query-syntax[Cloud Spanner SQL syntax] offers a variety of query hints to tune and optimize the performance of queries.
+If you find that you need to take advantage of this feature, you can achieve this in Hibernate using native SQL queries.
+
+This is an example of using the `@{FORCE_JOIN_ORDER=TRUE}` hint in a native Spanner SQL query.
+
+[source,java]
+----
+SQLQuery query = session.createSQLQuery("SELECT * FROM Singers AS s
+                                         JOIN@{FORCE_JOIN_ORDER=TRUE} Albums AS a
+                                         ON s.SingerId = a.Singerid
+                                         WHERE s.LastName LIKE '%x%'
+                                         AND a.AlbumTitle LIKE '%love%';");
+
+// Executes the query.
+List<Object[]> entities = query.list();
+----
+
+Also, you may consult the https://cloud.google.com/spanner/docs/sql-best-practices[Cloud Spanner documentation] on general recommendations for optimizing performance.
+
+== Cloud Spanner Hibernate ORM Limitations
+
+The Cloud Spanner Hibernate Dialect supports most of the standard Hibernate and Java Persistence annotations, but there are minor differences in supported features because of differences in Cloud Spanner from other traditional SQL databases.
 
 [options="header"]
 |===
 | Unsupported Feature | Description
+| Large DML Transactions | Each Spanner transaction may only have up to 20,000 operations which modify rows of a table.
 | Catalog and schema scoping for table names | Tables name references cannot contain periods or other punctuation.
-| Column default values | The dialect does not currently set default values based on the `@ColumnDefault` annotation, because Cloud Spanner does not support column defaults in the DDL.
+| Column default values | The dialect does not set default values based on the `@ColumnDefault` annotation, because Cloud Spanner does not support column defaults in the DDL.
 |===
 
+=== Limit Large DML Transactions
+
+Cloud Spanner has a mutation limit on each transaction - each Spanner transaction https://cloud.google.com/spanner/quotas#limits_for_creating_reading_updating_and_deleting_data[may only have up to 20,000 operations which modify rows of a table].
+
+NOTE: Deleting a row counts as one operation and inserting/updating rows will count as a number of operations equal to the number of affected columns.
```

Comment:
```suggestion
NOTE: Deleting a row counts as one operation and inserting/updating a single row will count as a number of operations equal to the number of affected columns.
```

Reply:
Over multiple rows, it would be numRows x numColumns, right? Might want to link to https://cloud.google.com/spanner/quotas#note2 or similar, since this is an intrinsically confusing aspect.

Reply:
Done.